### DISCUSSION TITLE
API works locally, but not in Docker Container

### DISCUSSION THREAD

**Discussion by FayZ676** [fastapi/fastapi#6115] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# DOCKERFILE 

FROM python:3.8
WORKDIR /app
RUN wget https://ofa-silicon.oss-us-west-1.aliyuncs.com/checkpoints/caption_large_best_clean.pt
RUN mkdir /app/checkpoints
RUN mv caption_large_best_clean.pt /app/checkpoints/caption.pt
COPY requirements.txt /app/requirements.txt
RUN pip install --no-cache-dir --upgrade -r /app/requirements.txt
COPY . /app/
CMD ["uvicorn", "main:app", "--port", "80", "--host", "0.0.0.0"]
```

### Description

I have a machine learning API that I am trying to deploy. It runs perfectly locally and I am able to successfully build a docker image of it using the command `docker build -t <image_name> .`. However, when I try to test to make sure the container works locally via the command `docker run -p 80:80 <image_name>` I get the output seen below. 

Given that the API uses a few machine learning libraries, I thought that perhaps the docker container is running out of memory and it is not a FastAPI problem, but I thought it worth a shot to ask here in the event that it actually is a problem with my FastAPI implementation.

Many Thanks

## Expected Output
```
2023-01-30 15:16:13 | INFO | fairseq.tasks.text_to_speech | Please install tensorboardX: pip install tensorboardX
2023-01-30 15:16:15 | INFO | tasks.ofa_task | source dictionary: 59457 types
2023-01-30 15:16:15 | INFO | tasks.ofa_task | target dictionary: 59457 types
/home/faizi/miniconda3/envs/rest-api/lib/python3.8/site-packages/torch/functional.py:504: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at ../aten/src/ATen/native/TensorShape.cpp:3190.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
/home/faizi/miniconda3/envs/rest-api/lib/python3.8/site-packages/torchvision/transforms/transforms.py:329: UserWarning: Argument 'interpolation' of type int is deprecated since 0.13 and will be removed in 0.15. Please use InterpolationMode enum.
  warnings.warn(
INFO:     Started server process [336953]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
```

## Actual Output
```
2023-01-30 23:12:56 | INFO | fairseq.tasks.text_to_speech | Please install tensorboardX: pip install tensorboardX
2023-01-30 23:13:00 | INFO | tasks.ofa_task | source dictionary: 59457 types
2023-01-30 23:13:00 | INFO | tasks.ofa_task | target dictionary: 59457 types
```

### Operating System

Linux

### Operating System Details

Ubuntu 22.04

### FastAPI Version

0.89.1

### Python Version

3.8.13

**Comment by PurtucRobert** [comment]
When running locally, are you building the env from the same requirements.txt as you're doing for the Docker image? 
Asking as I'm seeing this:
2023-01-30 23:12:56 | INFO | fairseq.tasks.text_to_speech | Please install tensorboardX: pip install tensorboardX

**Comment by mtaimoor1** [comment]
By looks of it everything looks fine. But there are some things that you can try which can maybe resolve your issue.

1. Try overriding the dockerfile CMD command in docker run using the following command
               `docker run -p 80:80 image_name 'uvicorn main:app --host 0.0.0.0 --port 80'`

2. Check and make sure that your local port is not pre-occupied by any other process

**Comment by FayZ676** [comment, marked as answer]
It turned out that `docker run` would end in the above response while `sudo docker run` would result in the expected output. I don't know why the absence of sudo permission would create such a response, but that seems to be the root of it at least. Thank you @PurtucRobert and @mtaimoor1 for your assistance.
